magma/interface.py

```diff
@@ -491,6 +494,12 @@ def __getattr__(self, key):
             return self._ports[key]
         return super().__getattribute__(key)
 
+    def flip(self):
+        return IO(**{
+            name: T.flip()
+            for name, T in zip(self._decl[::2], self._decl[1::2])
+        })
```

Comment:
if we add the function `fields`

```suggestion
        return IO(**{name: T.flip() for name, T in self.fields().items()})
```